### DISCUSSION TITLE
href-sanitizer

### DISCUSSION THREAD

**Discussion by ghajini** [uBlockOrigin/uBlock-issues#3487] [General] [answered]
On this site `https://zinkmovies.in/movies/download-johnny-johnny-yes-papa-dual-audio-movies/`

download now button has href like `href="https://modijiurl.com/full?api=1ef7670b4005f45d35ca1a9847ab4395c1455c2c&url=aHR0cHM6Ly9odWJjbG91ZC5hcnQvdmlkZW8vZWJ4MWxxYWxqY2FqZGtk&type=2"`
where &url attribute has base64 encoded link.

when `aHR0cHM6Ly9odWJjbG91ZC5hcnQvdmlkZW8vZWJ4MWxxYWxqY2FqZGtk` is decoded it resolves to `https://hubcloud.art/video/ebx1lqaljcajdkd` 
does href-sanitizer supports this decoding ?

 also annoyance can be avoided on those sites with `/safe.php?link=http$doc,urlskip=?link,domain=alljntuworld.in|creditcarred.com|crickethelp.in|mewatjobs.com|naamlist.com|newsonnline.com|phonesparrow.com|rajasthantopnews.com|ramrakshastotra.com|samaysuchi.com`

**Comment by ghajini** [comment]
looks like i can do `zinkmovies.in##+js(aopr, app_advert)` but can this be achieved with href-sanitizer?

**Comment by gorhill** [comment]
I could look into re-using the same `urlskip` code into `href-sanitizer`.

**Comment by ghajini** [comment]
another one `https://filmyvilla.site123.me/hollywood/venom-2018-full-movie-dual-audio-hdrip-720p`
![Untitled444](https://github.com/user-attachments/assets/08a64dca-de61-4fae-9ba2-911306c6dfb6)

**LINK1** has `href="https://filmyvilla.site123.me/versions/2/include/external_redirect.php?websiteID=1732528&url=https%3A%2F%2Fza.gl%2Ffull%2F%3Fapi%3D1d636cb7638ac24f99c7f291c0b0903e4f611216%26url%3DaHR0cHM6Ly91cHRvYm94LmNvbS8xdTZvcmkxZHI2bHU%3D%26type%3D2"`  => `https://za.gl/full/?api=1d636cb7638ac24f99c7f291c0b0903e4f611216&url=aHR0cHM6Ly91cHRvYm94LmNvbS8xdTZvcmkxZHI2bHU=&type=2`

**base64 part** `aHR0cHM6Ly91cHRvYm94LmNvbS8xdTZvcmkxZHI2bHU=` =>`https://uptobox.com/1u6ori1dr6lu`

**Reply by gorhill** [reply]
I get "404 Not Found" at `https://uptobox.com/1u6ori1dr6lu`. Is this expected?

**Reply by ghajini** [reply]
it is expected

**Reply by gorhill** [reply, marked as answer]
`urlskip=`-like syntax will be supported in next dev build: <https://github.com/gorhill/uBlock/commit/9bf8d53ebe7cab20ac8491741c5b99cf9d35d643>

**Reply by ghajini** [reply]
```
filmyvilla.site123.me##+js(href-sanitizer, a[href*="redirect"], ?url ?url -base64)
filmyvilla.site123.me##+js(href-sanitizer, a[href*="redirect"], ?url)
```
work
